addons/mail/static/src/messaging/component/message_list/message_list.js

```diff
@@ -276,6 +276,13 @@ class MessageList extends Component {
         ) {
             return false;
         }
+        if (
+            prevMessage.notifications.length > 0 ||
+            message.notifications.length > 0
+        ) {
+            // notifications are displayed per message
+            return false;
+        }
```

Comment:
This is needed because the `o_Message_header` is not displayed if messages are squashed, so it would be impossible to see the notifications from the 2nd message.

This is similar for moderation and all other conditions in this method.

Note that even functionally, I feel like emails should never be squashed (so maybe even expand this condition), as they are never sent in quick succession like chat messages, they all deserve their own header and stuff probably.

Reply:
Ok I see, let's keep like this then. Would clarify comment on why we don't want squash here, the current one is not clear.

Reply:
ok